test/PaymasterCommitment.test.ts

```diff
@@ -189,6 +189,53 @@ contract('Paymaster Commitment', function ([_, relayOwner, relayManager, relayWo
       assert.closeTo(paid, gasUsed, 50)
     })
 
+    it('paymaster should not pay for requests exceeding msg.data size limit', async () => {
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit))
+      await expectRevert(
+        relayHubInstance.relayCall(10e6, r.req, r.sig, hugeApprovalData, externalGasLimit, {
+          from: relayWorker,
+          gas: externalGasLimit,
+          gasPrice
+        }), 'msg.data exceeded limit'
+      )
+    })
+
+    it('paymaster should not pay for requests with max msg.data size if it rejects in pre', async () => {
+      await paymasterContract.setRevertPreRelayCall(true)
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit) - 1000)
```

Comment:
does it mean that the relay request (without the huge approval ) is 500 bytes long?

Reply:
my calculations (for OG-416) show a minimum (dynamic buffers empty) relayCall is 22 words (4+ 22*32=708 bytes)
every non-empty string adds 1-word length and actual data ( **signature** is always non-empty, **req.data** is almost always non-empty (at least from solidity calls), **paymasterData**, **approvalData** are optional)

bottom line: I'm not sure you got to this "1000"